app/scripts/modules/google/src/securityGroup/details/securityGroupDetail.controller.js

```diff
@@ -89,6 +89,20 @@ module.exports = angular
                 $scope.securityGroup.sourceTags = sourceTags.substring(1, sourceTags.length - 1).split(', ');
               }
 
+              // These come back from the global firewall endpoint as '[account-a@project.iam.gserviceaccount.com, account-b@project.iam.gserviceaccount.com]'
+              if (typeof $scope.securityGroup.targetServiceAccounts === 'string') {
```

Comment:
if these don't come back as a string, are they `undefined`? asking because if so we'll get an error when we try to read `.length` off them in the template

Reply:
@maggieneterval this block is basically a duplicate of the work done in the `sourceTags` and `targetTags` block above, so I'm not totally sure on this. I believe in the case of both tags and serviceAccounts if they don't exist they ARE undefined. Couple of examples in gate:

# Tag rule
```
{
id: "host/platform-vault",
moniker: {
app: "platform",
cluster: "platform-vault-egress-allow",
detail: "vault-egress-allow",
stack: "services"
},
name: "platform-vault-egress-allow",
network: "host/host-network",
selfLink: "https://www.googleapis.com/compute/beta/projects/prod-host/global/firewalls/platform-vault-egress-allow",
targetTags: "[vault-server]"
},
```

# Service Account rule
```
{
id: "host/platform-vault",
moniker: {
app: "platform",
cluster: "platform-vault-egress-allow",
detail: "vault-egress-allow",
stack: "services"
},
name: "platform-vault-egress-allow",
network: "host/host-network",
selfLink: "https://www.googleapis.com/compute/beta/projects/prod-host/global/firewalls/platform-vault-egress-allow",
targetServiceAccounts: "[vault@prod-app-vault.iam.gserviceaccount.com]"
}
```

Any suggestions on how to go forward would be great! Pretty new to this stuff :)

Reply:
Ahh I'm going to chalk this up to angular magic because you're right, this works fine even when each is undefined!